tests/rptest/tests/internal_topic_protection_test.py

```diff
@@ -0,0 +1,117 @@
+        def test_with_produce_fn(produce_fn, failure_exception_type):
+            test_topic = "noproduce_topic"
+            self.kafka_tools.create_topic_with_config(test_topic, 3, 3, {})
+
+            wait_until(lambda: test_topic in self.rpk.list_topics(),
+                       timeout_sec=90,
+                       backoff_sec=3)
+
+            # Ensure topic can't be produced to via the Kafka API when
+            # it's in the kafka_noproduce_topics list.
+            self.rpk.cluster_config_set("kafka_noproduce_topics",
+                                        str([test_topic]))
+
+            try:
+                produce_fn(test_topic, "test_msg")
+            except Exception as e:
+                assert type(e) == failure_exception_type
+            else:
+                assert False, "Call to delete topic returned sucess"
+
+            try:
+                self.rpk.consume(test_topic, n=1, timeout=30)
+            except RpkException:
+                pass
+            else:
+                assert False, "Messages exist in no produce topic"
+
+            # Check that a topic can be removed from the kafka_noproduce_topics
+            # list then produced to.
+            self.rpk.cluster_config_set("kafka_noproduce_topics", str([]))
+            produce_fn(test_topic, "test_msg")
+            self.rpk.consume(test_topic, n=1, timeout=30)
+
+            # Clean up the existing topic
+            self.rpk.delete_topic(test_topic)
+
+            wait_until(lambda: test_topic not in self.rpk.list_topics(),
+                       timeout_sec=90,
+                       backoff_sec=3)
+
+        test_with_produce_fn(
+            lambda topic, msg: self.kafka_cat.produce_one(topic, msg),
+            subprocess.CalledProcessError)
+        test_with_produce_fn(
+            lambda topic, msg: self.rpk.produce(topic, "key", msg, timeout=30),
+            RpkException)
+
+    @cluster(num_nodes=3)
+    def kafka_nodelete_topics_test(self):
+        def test_using_client(client, no_exception_on_delete):
+            test_topic = "nodelete_topic"
+            self.kafka_tools.create_topic_with_config(test_topic, 3, 3, {})
+
+            wait_until(lambda: test_topic in client.list_topics(),
+                       timeout_sec=90,
+                       backoff_sec=3)
+
+            # Ensure topic can't be deleted via the Kafka API when it's
+            # in the nodelete list.
+            self.rpk.cluster_config_set("kafka_nodelete_topics",
+                                        str([test_topic]))
+            try:
+                client.delete_topic(test_topic)
+            except subprocess.CalledProcessError:
+                pass
+            else:
+                assert no_exception_on_delete, "Call to delete topic returned sucess"
+
+            # allow time for any erronous deletion to be propagated
+            time.sleep(10)
+            assert test_topic in client.list_topics()
```

Comment:
```
wait_unit(lambda: test_topic in client.list_topic(), timeout_sec=10)
```
would something like this work instead of sleep + assert?

Reply:
The `time.sleep(10)` is to allow for a potential deletion to propagate throughout the cluster. `test_topic in client.list_topic()` should always be true so `wait_until` won't be too helpful here.